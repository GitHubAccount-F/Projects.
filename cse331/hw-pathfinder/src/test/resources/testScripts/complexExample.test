#Tests Dijkstra's Algorithm on a complex example

#Creates a graph
CreateGraph graph1

#Adds nodes
AddNode graph1 one
AddNode graph1 two
AddNode graph1 three
AddNode graph1 four
AddNode graph1 five
AddNode graph1 six

#Adds edges
AddEdge graph1 one two 4
AddEdge graph1 two one 4
AddEdge graph1 two one 1

AddEdge graph1 one three 4
AddEdge graph1 three one 4

AddEdge graph1 one four 4
AddEdge graph1 four one 4

AddEdge graph1 three five 4
AddEdge graph1 five three 4

AddEdge graph1 four five 10
AddEdge graph1 five four 10

#Finds the least costly paths
FindPath graph1 three six

FindPath graph1 five two

FindPath graph1 four five

FindPath graph1 two one

FindPath graph1 eight seven