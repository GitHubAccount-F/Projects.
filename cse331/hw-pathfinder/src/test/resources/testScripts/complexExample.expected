#Tests Dijkstra's Algorithm on a complex example

#Creates a graph
created graph graph1

#Adds nodes
added node one to graph1
added node two to graph1
added node three to graph1
added node four to graph1
added node five to graph1
added node six to graph1

#Adds edges
added edge 4.000 from one to two in graph1
added edge 4.000 from two to one in graph1
added edge 1.000 from two to one in graph1

added edge 4.000 from one to three in graph1
added edge 4.000 from three to one in graph1

added edge 4.000 from one to four in graph1
added edge 4.000 from four to one in graph1

added edge 4.000 from three to five in graph1
added edge 4.000 from five to three in graph1

added edge 10.000 from four to five in graph1
added edge 10.000 from five to four in graph1

#Finds the least costly paths
path from three to six:
no path found

path from five to two:
five to three with weight 4.000
three to one with weight 4.000
one to two with weight 4.000
total cost: 12.000

path from four to five:
four to five with weight 10.000
total cost: 10.000

path from two to one:
two to one with weight 1.000
total cost: 1.000

unknown: eight
unknown: seven